planet/http.py

```diff
@@ -272,6 +275,20 @@ async def alog_response(*args, **kwargs):
         self._limiter = _Limiter(rate_limit=RATE_LIMIT, max_workers=MAX_ACTIVE)
         self.outcomes: Counter[str] = Counter()
 
+        # create a dedicated event loop for this httpx session.
+        def _start_background_loop(loop):
+            asyncio.set_event_loop(loop)
+            loop.run_forever()
+
+        self._loop = asyncio.new_event_loop()
+        self._loop_thread = threading.Thread(target=_start_background_loop,
+                                             args=(self._loop, ),
+                                             daemon=True)
+        self._loop_thread.start()
```

Comment:
What does this do?

Reply:
Since the methods of the sync client generally just call out to the async client methods, we need to use those async methods within an event loop. An easy way to do this is to invoke the methods using `asyncio.run` or `asyncio.get_running_loop`, however one issue with this approach is that the httpx session (which is coupled to our own Session class) is tied to a single event loop. If we close the event loop, our Session instance will no longer be usable.

Creating a new event loop in a thread is a way to ensure that the httpx session will have a persistent loop for as long as our sdk's Session is alive. This lets us manage the asyncio loop (and the Session instance) behind the scenes without the user needing to as they would with the async client.

Reply:
Also, you won't hear me argue that this is an ideal long term solution. I think it is a good step in the right direction for users asking for an easier to use client.

If adoption of the sync client is promising, it may be worthwhile to transition the sync client to have its own non-async http session with no event loop in the picture. Something like @ischneider's idea of shared business logic and request setup, with either sync or async request dispatching (and my preference would be to abstract all that from users). We might be able to use the Features and/or Basemaps clients to test some ideas out.

Reply:
Thanks for explaining!  I understand why this is needed in the current state of the world, and agree that maybe in the future when we have less restraints for backwards compatibility that we could try another approach.  Otherwise  for now this is a clever solution that will help ship the sync client sooner rather than later!